### PR TITLE
Import `lazy_loader` as private symbol in top-level namespaces

### LINKED ISSUE #7539: Polluted namespace by lazy loader
### Description:

In the `__init__.py` files, the `lazy_loader` has been used but it's import hasn't been deleted which is somewhat polluting the module namespace. It may be useful to remove it.

### Way to reproduce:

```python
import skimage

print(skimage.lazy, skimage.color.lazy)
```

### Version information:

```Shell
3.9.19 (main, May  6 2024, 20:12:36) [MSC v.1916 64 bit (AMD64)]
Windows-10-10.0.22631-SP0
scikit-image version: 0.24.0
numpy version: 2.0.2
```

### PR DESCRIPTION
**Pull request by kushaangupta**
## Description

Closes #7539

### CODE REVIEW

**Comment by stefanv** [comment]
Thanks for the contribution! 

I think fine to rename to use the underscore (easiest), or to del. Both aren't necessary?

**Comment by kushaangupta** [comment]
Thanks for the feedback @stefanv ! I used the precedent of napari in which they use both underscore before and delete the function afterwards. My rationale was to use underscore was just so nothing in submodules gets overwritten and deletion would ensure efficient memory usage. I can edit to either, if you prefer.

**Comment by stefanv** [comment]
Really, it's probably fine as it is as well. Let's let a second reviewer merge.

**Review by lagru — COMMENTED** [review]
Personally, I'm not sure this change is really necessary. 

While accessing `skimage.lazy` is possible, most users should never discover it? It doesn't appear in `dir(skimage)`, and neither does it with tab completion. It only appears in `skimage.__dict__`.

That said I'm happy to merge the refactor of `lazy` to `_lazy`. It's the more expressive name.

File: skimage/__init__.py [hunk 1]
```diff
@@ -67,9 +67,11 @@
 
 __version__ = '0.24.1rc0.dev0'
 
-import lazy_loader as lazy
+import lazy_loader as _lazy
 
-__getattr__, __lazy_dir__, _ = lazy.attach_stub(__name__, __file__)
+__getattr__, __lazy_dir__, _ = _lazy.attach_stub(__name__, __file__)
+
+del _lazy
```
[the remark this answers is not in the document]
Comment (kushaangupta) [reply skimage/__init__.py]: Thanks! Ease of debugging seems a fair reason to keep the imported modules. I've removed the `del` statements.

**Review by lagru — APPROVED** [review]
Looks good. Thanks! I'm merging since @stefanv implicitly approved in https://github.com/scikit-image/scikit-image/pull/7540#issuecomment-2350773897.

**Pull request merged**